benchmarks_test.go

```diff
@@ -22,7 +22,12 @@ func BenchmarkOneToOnePoller(b *testing.B) {
 		panic("Oops...")
 	})))
 
+	var wg sync.WaitGroup
+	wg.Add(1)
+	defer wg.Wait()
+
 	go func() {
+		defer wg.Done()
```

Comment:
It seems weird to wait on this function to return. `d.Next()` will block until a value is ready to be read, and if you read `b.N` times then you will have to have written `b.N` times, which means all writes have completed. I don't know what you are hoping to measure in addition to the write times, perhaps `WaitGroup` and goroutine overhead?